source/deploy/postgres-migration.rst

```diff
@@ -6,32 +6,167 @@ Migration guidelines from MySQL to PostgreSQL
+Our ``migration-assist`` tool provides an efficient, error-free migration experience that automates the tasks to be executed. The tool offers 3 core utility commands:
+
+1. ``migration-assist mysql``
+
+   Checks the MySQL database schema to ensure readiness for migration, and offers fixes for common issues.
+
+2. ``migration-assist postgres``
+
+   Creates the PostgreSQL database schema, and prepares it for Mattermost deployment by downloading the necessary migrations and applying them.
+
+3. ``migration-assist pgloader``
+
+   Generates a pgloader configuration from DSN values, ensuring accurate data transfer.
+
+Install
+~~~~~~~
+
+The ``migration-assist`` tool can be downloaded and compiled with the `Go <https://go.dev/>`__ toolchain. The tool requires at least ``v1.22`` of the Go compiler.
+
+Use ``go install`` to install the tool:
+
+.. code-block:: shell
+
+   go install github.com/mattermost/migration-assist/cmd/migration-assist@latest
+
+Usage
+~~~~~~
+
+.. important::
+
+   Please make sure you have the necessary environment to perform the migration. Ensure that the MySQL and PostgreSQL databases are running and accessible. To setup a PostgreSQL instance, see the :doc:`database </install/prepare-mattermost-database>` documentation for details.
+
+Step 1 - Check the MySQL database schema
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Run the following command to check the MySQL database schema:
+
+.. code-block:: shell
+
+   migration-assist mysql "<MYSQL_DSN>"
+
+This command outputs the readiness status and prints required fixes for common issues. The flags for fixes are as follows (where all fixes can be used together at the same time):
+
+.. code-block:: shell
+
+   --fix-artifacts               Removes the artifacts from older versions of Mattermost
+   --fix-unicode                 Removes the unsupported unicode characters from MySQL tables
+   --fix-varchar                 Removes the rows with varchar overflow
+
+Step 2 - Create the PostgreSQL database schema
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Run the following command to create the Postgres database schema:
+
+.. code-block:: shell
+
+   migration-assist postgres "<POSTGRES_DSN>" --run-migrations --mattermost-version="<MATTERMOST_VERSION>"
+
+This command downloads the necessary migrations and applies them to the Postgres database. The ``--mattermost-version`` flag is required to specify the Mattermost version you are migrating from.
+
+Step 3 - Generate a pgloader configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Run the following command to generate a pgloader configuration:
+
+.. code-block:: shell
+
+   migration-assist pgloader --mysql="<MYSQL_DSN>" --postgres="<POSTGRES_DSN>" > migration.load
+
+This command will generate a pgloader configuration file that can be used to migrate the data from MySQL to Postgres.
+
+Step 4 - Run pgloader
+^^^^^^^^^^^^^^^^^^^^^^
+
+See the `install pgloader <#install-pgloader>`__ section for installation details.
```

Comment:
Again, maybe a teaser that pgloader will come in the `migraton-assist.zip` (build for Ubuntu/Debian).

Again, your thoughts @isacikgoz.

Reply:
I think the docker image should cover that and maybe it's best to hold on for that (for now).